blacklist/blacklist/Scenes/Home/HomeViewController.swift

```diff
@@ -8,24 +8,172 @@
 
 import UIKit
 
-class HomeViewController: UIViewController {
-    fileprivate let presenter = HomePresenter()
+class HomeViewController: UIViewController, ViewControllerProtocol {
+    typealias P = HomePresenter
+    typealias R = HomeRouter
+
+    var presenter: HomePresenter!
+
+    struct Constants {
+        static let upcomingTableViewCell = "UpcomingTableViewCell"
+    }
+
+    fileprivate var tableViewTopConstraintConstant: CGFloat = 0.0
+
+    // IBOutlets
+    @IBOutlet weak var lendingAmountLabel: UILabel!
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var gradientContainerView: UIView!
+    @IBOutlet weak var tableViewTopConstraint: NSLayoutConstraint!
+    @IBOutlet weak var maximumUpcomingHuggingConstraint: NSLayoutConstraint!
 
     override func viewDidLoad() {
+        // Let's configure the presenter
         super.viewDidLoad()
 
-        // Let's configure the presenter
-        presenter.attach(view: self)
-        presenter.register(router: HomeRouter(viewController: self))
+        configure { (context) -> P! in
+            let presenter = HomePresenter()
+            presenter.router = HomeRouter(viewController: context)
+
+            return presenter
+        }
+
+        configure()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        // Add a bounce effect to the first cell
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) { [weak self] in
+            if let tableView = self?.tableView, tableView.visibleCells.count > 0 {
+                if let cell = self?.tableView.visibleCells.first as? UpcomingTableViewCell {
+                    cell.bounce()
+                }
+            }
+        }
     }
 
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         presenter.router?.prepare(for: segue, sender: sender)
     }
+
+    override func viewDidLayoutSubviews() {
+        // Let's create the gradient layer
+        let gradient = CAGradientLayer()
+        gradient.frame = gradientContainerView.bounds
+        gradient.colors = [UIColor.black.cgColor, UIColor.clear.cgColor]
+        gradient.locations = [0.9, 1.0]
+
+        gradientContainerView.layer.mask = gradient
+
+        super.viewDidLayoutSubviews()
+    }
+
+    func configure() {
+        configureTableView()
+
+        self.automaticallyAdjustsScrollViewInsets = false
+    }
 }
 
+// MARK: - TableView DataSource
+
+extension HomeViewController: UITableViewDataSource, UITableViewDelegate {
+    func configureTableView() {
+        // Let's configure the tableView
+        let upcomingTableViewCellNib = UINib(nibName: Constants.upcomingTableViewCell, bundle: nil)
+        tableView.register(upcomingTableViewCellNib, forCellReuseIdentifier: Constants.upcomingTableViewCell)
+
+        tableView.estimatedRowHeight = 68.0
```

Comment:
This should be a constant

Reply:
I'll make those changes and update the PR